src/App.tsx

```diff
@@ -17,18 +47,36 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                onChange={setQuery}
+                sortValue={sort}
+                onSort={setSort}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!todos.length && <Loader />}
+
+              <TodoList
+                todos={filteredTodos}
+                setTodo={setSelectedTodo}
+                todoModal={todoModal}
+                setTodoModal={setTodoModal}
+              />
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {
+        todoModal && (
+          <TodoModal
+            selectedTodo={selectedTodo}
+            setSelectedTodo={setSelectedTodo}
+            setTodoModal={setTodoModal}
```

Comment:
just make 1 function for it
```suggestion
            onClose={handleCloseModal}
```